### PR TITLE
Add continuous deployment

### PR DESCRIPTION
**Pull request by liconway**
Adds continuous deployment to the website.

Any push to the `master` branch triggers this job. The job is triggered in two stages:

1) Build
The code is built using NodeJS 14.x. If the build is successful, it is archived.

2) Deploy
Deploy uses the previous archive, and uploads the contents to the `gh-pages` branch.

### CODE REVIEW

**Review by JessieG123 — APPROVED** [review]
I trust you because I honestly don't know how this works lol

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          persist-credentials: false
+      - name: Download build
+        uses: actions/download-artifact@v1
+        with:
+          name: deploy_dist
+      - name: Deploy to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bill-luu) [inline .github/workflows/deploy.yml:42]: Where are the secrets? 👀

**Comment by bill-luu** [comment]
I see a lot of deletions hope those aren't important 😬

**Comment by liconway** [comment]
> I see a lot of deletions hope those aren't important 😬

We were manually building the site and storing them in a directory, which is a very bad practice. Since this CD now builds the files on it's own, the manual files are no longer needed and can be deleted.

Made this PR pretty quick so should've explained what I was doing better lol

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (liconway) [reply .github/workflows/deploy.yml:42]: Not 100% certain, but afaik it's the GitHub Access Token for whoever pushed the commit to master.

This action takes the previously built files and pushes them to the `gh-pages` branch. You can see this below:
https://github.com/UCSolarCarTeam/UCSolarCarTeam.github.io/tree/gh-pages

Notice how the latest commit shows that I was the author. I technically was but I never created the commit, as it was automatically created by this action.

**Review by JaydenBischoff — APPROVED** [review]

**Comment by liconway** [comment]
Thanks everyone! Merging this now

**Pull request merged**